tests/core/test_stellargraph.py

```diff
@@ -535,3 +535,71 @@ def f():
         return StellarGraph(g, node_features=node_features)
 
     allocation_benchmark(f)
+
+
+def example_weighted_hin():
+    graph = nx.MultiDiGraph()
+    graph.add_nodes_from([0, 1], label="A")
+    graph.add_nodes_from([2, 3], label="B")
+    graph.add_edges_from([(0, 1, 0.0), (0, 1, 1.0)], label="AA")
```

Comment:
The `0.0` here is being used as a multigraph edge key, not as a weight. I think this should be:

```suggestion
    graph.add_edge((0, 1), label="AA", weight=0.0)
    graph.add_edge((0, 1), label="AA", weight=1.0)
```

See https://networkx.github.io/documentation/stable/reference/classes/generated/networkx.MultiGraph.add_edges_from.html#networkx.MultiGraph.add_edges_from . In particular

> The edges can be:
> - ...
> - 3-tuples (u, v, k) for not iterable key k

and also https://networkx.github.io/documentation/stable/reference/classes/multigraph.html

> If an edge already exists, an additional edge is created and stored using a key to identify the edge. By default the key is the lowest unused integer.


- [ ] and similarly `weight=10.0` instead of the tuple in the next line.

Reply:
Ah nice catch, I meant to use https://networkx.github.io/documentation/networkx-2.2/reference/classes/generated/networkx.MultiDiGraph.add_weighted_edges_from.html#networkx.MultiDiGraph.add_weighted_edges_from here actually